src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -153,48 +158,6 @@ String determineSwitchOnSize(Long sizeOfRepo, GitTool tool) {
         return "NONE";
     }
 
-    /**
-     * For a given recommended git implementation, validate if the installation exists and provide no suggestion if
-     * implementation doesn't exist.
-     * @param gitImplementation: The recommended git implementation, "git" or "jgit" on the basis of the heuristics.
-     */
-    private void determineGitTool(String gitImplementation) {
-        if (gitImplementation.equals("NONE")) {
-            gitTool = "NONE";
-            return; // Recommend nothing (GitToolRecommendation = NONE)
-        }
-        final Jenkins jenkins = Jenkins.get();
-        GitTool tool = GitUtils.resolveGitTool(gitImplementation, jenkins, null, TaskListener.NULL);
-        if (tool != null) {
-            gitTool = tool.getGitExe();
-        }
-    }
-
-    private GitTool getResolvedGitTool(String recommendation) {
-        final Jenkins jenkins = Jenkins.get();
-        return GitUtils.resolveGitTool(recommendation, jenkins, null, listener);
-    }
-
-    /**
-     * Recommend git tool to be used by the git client
-     * @return git implementation recommendation in the form of a string
-     */
-    public String getGitTool() {
-        return gitTool;
-    }
-
-    private String determineToolName(String gitExe, String recommendation) {
-        if (gitExe.contains(recommendation) && !gitExe.equals(JGitTool.MAGIC_EXENAME) && !gitExe.equals(JGitApacheTool.MAGIC_EXENAME)) {
-            return gitExe;
-        }
-        if (!recommendation.equals(gitExe)) {
-            if (gitExe.equals(JGitApacheTool.MAGIC_EXENAME) && recommendation.equals(JGitTool.MAGIC_EXENAME)) {
-                return gitExe;
-            }
-        }
-        return recommendation;
-    }
-
     private GitTool resolveGitToolForRecommendation(GitTool userChoice, String recommendation) {
         GitTool tool;
         if (recommendation.equals(JGitTool.MAGIC_EXENAME)) {
```

Comment:
For repository size < 5MiB, the recommendation should be `jgit/jgitapache`. This branch of logic will try to recommend `jgit or jgitapache` and confirm with the system if it is enabled by the user. It would return a null if it is not enabled by the user.